var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -56,9 +58,12 @@ class Caliper(CMakePackage):
     # gotcha doesn't work on Mac
     variant('gotcha', default=sys.platform != 'darwin',
             description='Enable GOTCHA support')
+    variant('sampler', default=sys.platform == 'linux2',
+            description='Enable sampling support on Linux')
     variant('sosflow', default=False,
             description='Enable SOSflow support')
 
+    depends_on('gotcha@1.0:', when='+gotcha')
```

Comment:
If 1.6 downloads the wrong gotcha version, why not:

```python
depends_on('gotcha@<something>', when='@:1.6')
depends_on('gotcha@1.0:', when='@1.7:')
```
?

Reply:
Nevermind, saw the comment on #8558.  Just remove 1.6.0 if it downloads its own dependencies.